jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -82,80 +77,71 @@ protected HttpClient getHttpClient()
         // Return new instances every time to keep track of the response content
         return new AuthenticationListener();
     }
-    
 
-    
-    protected List<HeaderInfo> getHeaderInfo(String value) throws IllegalArgumentException
+
+    protected List<HeaderInfo> getHeaderInfo(String header) throws IllegalArgumentException
     {
-        String header = value;
         List<HeaderInfo> headerInfos = new ArrayList<>();
-        
-        while(true)
+
+        Matcher m;
+        String authScheme = null;
+        Map<String,String> authParams = new HashMap<>();
+
+        for(String value : new QuotedCSV(true, header))
         {
-            Matcher m = MULTIPLE_CHALLENGE_PATTERN.matcher(header);
+            m = CHALLENGE_PATTERN.matcher(value);
             if (m.matches())
             {
-                headerInfos.add(newHeaderInfo(m.group(1)));
-                header = m.group(2);
-            }
-            else
-            {
-                headerInfos.add(newHeaderInfo(header));
-                break;
+                if(m.group("schemeOnly") != null)
+                {
+                    if (authScheme != null)
+                    {
+                        headerInfos.add(new HeaderInfo(getAuthorizationHeader(), authScheme, authParams));
+                        authParams = new HashMap<>();
+                    }
+
+                    authScheme = m.group(1);
+                    continue;
+                }
+
+                if (m.group("scheme") != null)
+                {
+                    if (authScheme != null)
+                    {
+                        headerInfos.add(new HeaderInfo(getAuthorizationHeader(), authScheme, authParams));
+                        authParams = new HashMap<>();
+                    }
+
+                    authScheme = m.group("scheme");
+                }
+
+                if (authScheme == null)
+                    throw new IllegalArgumentException("Parameters without auth-scheme");
+
+                if (m.group("paramName") != null)
+                {
+                    String paramVal = (m.group("paramValueQuoted") != null) ? m.group("paramValueQuoted") : m.group("paramValueUnquoted");
```

Comment:
A quoted string can contain escaped characters, so you can't just get the value without quotes.  You need to pass it to the `unquote` method (on one of our utility classes... can't remember which is the right one).   As it checks for quotes and just returns the string if it is not quoted, you may as well just always pass it the value.